### ISSUE TITLE
`custom-error-definition`: false positive on error classes extending a non-Error base

### ISSUE THREAD

**Issue by lexa-uw** [sindresorhus/eslint-plugin-unicorn#3614]
The rule requires `options` as the second constructor parameter and forwards it to `super()` as `ErrorOptions`. It applies this to every class whose name ends with `Error`, including classes extending a framework base whose second constructor argument is **not** `ErrorOptions`.

The most common case is `GraphQLError` from graphql, whose signature is `constructor(message: string, options?: GraphQLErrorOptions)` where 
`GraphQLErrorOptions` is `{ nodes, source, positions, path, originalError, extensions }` — a different type that does not accept cause.

```js
import { GraphQLError } from 'graphql'

export class UserNotFoundError extends GraphQLError {
  public constructor(userId: string) {
    super(`User "${userId}" does not exist`, { extensions: { code: 'NOT_FOUND' } })
    this.name = 'UserNotFoundError'
  }
}
// error: Error constructors should accept `options` as the second parameter.
```

Following the rule here produces a wrong signature: `options?: ErrorOptions` cannot be forwarded to a `super()` that expects `GraphQLErrorOptions`, and the `{ extensions }` argument already occupies that slot.

Suggested fix: skip classes whose `superClass` is not `Error` or a known native error subclass; or add an option such as `ignoredBaseClasses: string[]` (default `[]`) so consumers can opt classes out by base name.

Version: eslint-plugin-unicorn 73.0.0
